beacon-chain/forkchoice/doubly-linked-tree/unrealized_justification.go

```diff
@@ -56,17 +58,26 @@ func (f *ForkChoice) updateUnrealizedCheckpoints() {
 		if node.justifiedEpoch > f.store.justifiedCheckpoint.Epoch {
 			f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
 			f.store.justifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
+			if err := f.updateJustifiedBalances(ctx, f.store.justifiedCheckpoint.Root); err != nil {
+				return errors.Wrap(err, "could not update justified balances")
+			}
 			if !features.Get().EnableDefensivePull && node.justifiedEpoch > f.store.bestJustifiedCheckpoint.Epoch {
 				f.store.bestJustifiedCheckpoint = f.store.unrealizedJustifiedCheckpoint
 			}
 		}
 		if node.finalizedEpoch > f.store.finalizedCheckpoint.Epoch {
 			if !features.Get().EnableDefensivePull {
+				if f.store.unrealizedJustifiedCheckpoint.Epoch > f.store.justifiedCheckpoint.Epoch {
+					if err := f.updateJustifiedBalances(ctx, f.store.unrealizedJustifiedCheckpoint.Root); err != nil {
+						return errors.Wrap(err, "could not update justified balances")
+					}
+				}
```

Comment:
We always update justified checkpoints but refuse to update balances unless the unrealized epoch is greater. Seems asymmetric. Can you explain the rationale here?

Reply:
Yes this can be made symmetric, but better use a != instead of > here.